src/Type/Integer.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Integer value object class.
+ *
+ * Implemented as an immutable object with a pair of named constructors.
+ */
+
+namespace Joomla\Webservices\Type;
+
+final class Integer extends Type
```

Comment:
Now that's what I call a powerful argument. :-)

Looks like I need to rename Float and String too.